test/TBD/async-function-pointer.swift

```diff
@@ -0,0 +1,5 @@
+// REQUIRES: VENDOR=apple 
+// RUN: %target-swift-frontend -emit-ir -o/dev/null %s -enable-experimental-concurrency -validate-tbd-against-ir=all
```

Comment:
It doesn't harm to also do a FileCheck on the generated IR file to check the linkage of the function pointer.

Reply:
Added.  That is also tested in many of the tests in IRGen/async.